src/Updown.ts

```diff
@@ -11,7 +11,7 @@ export interface Options {
 
 /** An object with primitive values (i.e. no arrays/objects) */
 interface PrimitiveObject {
-	[key : string] : string | number;
+	[key: string]: string | number;
```

Comment:
> @woubuc:
> There should always be a space before and after the : for type notations, as in the original code. This is to more easily see the difference between `{ prop : type }` and `{ prop: value }` even out of context.

It's more commonly used this way instead of the way you're saying if you look at other (big) projects.

Reply:
"Other people do x" on its own is not a valid reason to also do *x* imo. Rather, it can lead to a reason. Can you tell me *why* they do it that way? What's the benefit of writing it the way you suggest?

Reply:
They all use the recommended ESLint rules (by the ESLint team itself), which is imo a valid reason since they've seen the way it's used and looks best I think.

Reply:
@woubuc I've looked into this and setting the `@typescript-eslint/type-annotation-spacing` rule will fix this:

```ts
'@typescript-eslint/type-annotation-spacing': ['error', { before: true }],
```

The only advantage you'll have is that it ESLint will now add a space between the questionmark (`?`) and colon (`:`) too:

```diff
- public async getMetrics(token : string, from ?: Date, to ?: Date, group : 'time' | 'host' = 'host') {
+ public async getMetrics(token : string, from ? : Date, to ? : Date, group : 'time' | 'host' = 'host') {
```

Reply:
That's strange. I guess it's meant to be `from? : Date`, or does it enforce the space before the `?` as well?

If it's allowed by that rule, `from? : Date` is acceptable to me.

Reply:
Oh my bad.
The diff should be
```diff
- public async getMetrics(token : string, from? : Date, to? : Date, group : 'time' | 'host' = 'host') {
+ public async getMetrics(token : string, from ? : Date, to ? : Date, group : 'time' | 'host' = 'host') {
```

Reply:
If I do it like

```ts
public async getMetrics(token : string, from ?: Date, to ?: Date, group : 'time' | 'host' = 'host') {
```

ESLint won't complain anymore

Reply:
That's a pretty weird construction. Can you figure out if there's a way to target the `?` symbol specifically to resolve this syntax?

Reply:
Ah, I didn't see your latest comment. `from ?: Date` is fine.